### ISSUE TITLE
add git lfs support

### ISSUE THREAD

**Issue by jbicket** [buildkite/elastic-ci-stack-for-aws#467]
Can you include git-lfs in the base image?

https://git-lfs.github.com/

Both gitlab and github support it for their repos out of the box.

I've used this in Dockerfiles to install it:
```
wget -q https://github.com/git-lfs/git-lfs/archive/v2.4.0.tar.gz \
    && tar -zxf v2.4.0.tar.gz  \
    && cd git-lfs-2.4.0/ \
    && ./script/bootstrap  \
    && mv bin/git-lfs /usr/bin/ \
    && git lfs install
```
which if I were writing the PR I think can just go in one of your packer/scripts/install-gitlfs.sh, and then update buildkite-ami.json.

**Comment by lox** [comment]
Would be happy to accept a PR @jbicket! Otherwise we'll look at it when we get a chance!

**Comment by jbicket** [comment]
I can't test it in your codebase, verify the ami change is correct, etc.

**Comment by lox** [comment]
No problems, will do my best to find some time to make the change!

**Comment by lox** [comment]
Like now for instance!

**Comment by jbicket** [comment]
🙌 🔥 🙌 😄 🎊

**Comment by jbicket** [comment]
Obviously, we're happy to test it and verify it works/report back once your pr is merged :)

**Comment by jbicket** [comment]
I was able to verify this worked for us.  This is great, thanks @lox !

You might have a better way to do this than we figured out, but we added the following hooks to get it to work so that the git lfs fetch was fast(er) (i.e. it can fetch the objects in parallel instead of serially during a git clone). 
```
diff --git a/.buildkite/hooks/post-checkout b/.buildkite/hooks/post-checkout
new file mode 100755
index 0000000..55f227f
--- /dev/null
+++ b/.buildkite/hooks/post-checkout
@@ -0,0 +1,3 @@
+#!/bin/bash
+git lfs install
+git lfs pull

diff --git a/.buildkite/hooks/environment b/.buildkite/hooks/environment
new file mode 100755
index 0000000..0d12274
--- /dev/null
+++ b/.buildkite/hooks/environment
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+export GIT_LFS_SKIP_SMUDGE=1

```

**Comment by lox** [comment]
No problems!

Are those hooks in your repository? If so, that environment hook won't work. Check out https://buildkite.com/docs/agent/v3/hooks, the `environment` hook is an Agent level hook. You can set it in an `env` file in your secrets bucket and it should work though. See https://github.com/buildkite/elastic-ci-stack-for-aws#build-secrets for more context.

Beyond that, I think you can skip `git lfs install`, as it's done globally as part of the lfs install!
